src/main/java/com/vmware/vipclient/i18n/messages/service/LocaleService.java

```diff
@@ -31,119 +34,166 @@ public LocaleService(LocaleDTO dto) {
     }
 
     public Map<String, String> getRegions(String locale){
-        Map<String, String> regionMap = getRegionsByLocale(locale);
-        if (regionMap != null) {
-            return regionMap;
+        LocaleCacheItem cacheItem = getRegionsByLocale(locale);
+        if (!cacheItem.getCachedData().isEmpty()) {
+            return cacheItem.getCachedData();
         }
         Iterator<Locale> fallbackLocalesIter = LocaleUtility.getFallbackLocales().iterator();
         while (fallbackLocalesIter.hasNext()) {
             String fallbackLocale = fallbackLocalesIter.next().toLanguageTag();
             if(fallbackLocale.equalsIgnoreCase(locale))
                 continue;
             logger.info("Can't find regions for locale [{}], look for fallback locale [{}] regions as fallback!", locale, fallbackLocale);
-            regionMap = getRegionsByLocale(fallbackLocale);
-            if (regionMap != null) {
-                new FormattingCacheService().addRegions(locale, regionMap);
-                logger.debug("Fallback locale [{}] regions is cached for locale [{}]!\n\n", fallbackLocale, locale);
+            cacheItem = getRegionsByLocale(fallbackLocale);
+            if (!cacheItem.getCachedData().isEmpty()) {
                 break;
             }
         }
-        return regionMap;
+        return cacheItem.getCachedData();
     }
 
-    public Map<String, String> getRegionsByLocale(String locale){
+    public LocaleCacheItem getRegionsByLocale(String locale){
```

Comment:
```suggestion
    public LocaleCacheItem getRegionsFromCache(String locale){
```

Reply:
This method is for getting regions of specified locale not only from cache, but also from data source, so it not good to change to 'getPatternsFromCache'.